mlir/lib/Conversion/MemRefToEmitC/MemRefToEmitCPass.cpp

```diff
@@ -33,12 +34,13 @@ struct ConvertMemRefToEmitCPass
 
     // Fallback for other types.
     converter.addConversion([](Type type) -> std::optional<Type> {
-      if (isa<MemRefType>(type))
+      if (isa<MemRefType, IndexType>(type))
         return {};
       return type;
```

Comment:
```suggestion
      if (emitc::isSupportedEmitCType(type))
        return type;
      // Fail to convert unsupported types.
      return {};
```

Reply:
Ah, I think this can even just use `populateEmitCDefaultTypeConversions`

Reply:
I went your route instead, `populateEmitCDefaultTypeConversions` was confusing as one could take it to a function that actually added non-trivial conversions.